Automattic/woocommerce-payments
# Show notices on currencies that are bound to an UPE method

Fixes #2682

#### Changes proposed in this Pull Request



This PR changes how the delete button on the enabled currency list works. 

There are two scenarios in this approach: 

1. Removing a currency that has no bound UPE method

The delete button removing the currency will delete the currency instantly from the list without showing any modals, alerts, warnings etc. 

2. Removing a currency that has one or more bound UPE methods to itself

This type of currencies are deleted with a confirmation modal, showing the bound methods that will be obsolete after removing the currency:

![image](https://user-images.githubusercontent.com/3295/133831873-3bdd43c8-e82a-4ca0-8380-5d3f6a996e65.png)

#### Testing instructions





##### Automated testing

* The automated tests are adapted to this change, and added a new test for the first scenario (which directly deletes the currency)

##### Manual testing

- Go to Payment methods settings page, enable some of the payment methods which require an additional currency to operate (e.g Sofort, giropay requires EUR)
- Go to Multi-currency settings page, try to remove EUR from the list, assert that the modal is shown, and unless you click the Remove button, the currency is not removed.
- Assert that the texts are correct, it's showing the right symbol, right cards for the payment methods which will get obsolete, right title
- Then, try to delete an independent currency, assert that the modal isn't shown and the currency is immediately deleted.

-------------------


**Post merge**
